src/common/collections.rs

```diff
@@ -56,11 +83,27 @@ pub async fn do_get_collection(
     Ok(collection.info(&shard_selection).await?)
 }
 
-pub async fn do_list_collections(toc: &TableOfContent) -> CollectionsResponse {
+pub async fn do_list_collections(
+    toc: &TableOfContent,
+    claims: Option<Claims>,
+) -> CollectionsResponse {
+    let claims_collections = match claims.as_ref() {
+        Some(Claims {
+            exp: _,
+            w: _,
+            collections,
+            payload: _,
+        }) => collections.as_ref(),
+        None => None,
+    };
+
     let collections = toc
         .all_collections()
         .await
         .into_iter()
+        .filter(|c| {
+            claims_collections.map_or(true, |claims_collections| claims_collections.contains(c))
+        })
```

Comment:
I could see these convenience reasons to allow `GET /collections` with a `collection` claim:

1. To check which of the allowed collections exist. (e.g. claims=`{"collections":["a","b","c"]}`, but only `a` and `c` exist).
   This would be more relevant if/after we add wildcard support for the `collection` claim. (e.g. `{"collections": ["foo_*"]}`).

2. For automation purposes.
   A user code could iterate over all collections listed in the `GET /collections` response regardless of whether it has api-key or not.
   An alternative would be to decode api-key in one branch and to call `GET /collections` in another branch, which is less convenient for the user.


Would allowing this imply security or any other concerns?

Reply:
Solid points, specially the 1st one. I don't have any security concern about this 👍